plugins/kinetic-dp/fu-kinetic-dp-aux-isp.c

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (C) 2021 Jeffrey Lin <jlin@kinet-ic.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "fu-kinetic-dp-aux-isp.h"
+
+#include <fwupdplugin.h>
+
+#include "fu-kinetic-dp-aux-dpcd.h"
+#include "fu-kinetic-dp-secure-aux-isp.h"
+
+typedef struct {
+	KtChipId root_dev_chip_id;
+	KtFwRunState root_dev_state;
+} FuKineticDpAuxIspPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE(FuKineticDpAuxIsp, fu_kinetic_dp_aux_isp, G_TYPE_OBJECT)
+#define GET_PRIVATE(o) (fu_kinetic_dp_aux_isp_get_instance_private(o))
+
+typedef struct {
+	KtChipId chip_id;
+	KtFwRunState fw_run_state;
+	guint8 id_str[DPCD_SIZE_BRANCH_DEV_ID_STR];
+	guint8 str_len;
+} KtDpChipBrIdStrTable;
+
+/* Kinetic chip DPCD branch ID string table */
+static const KtDpChipBrIdStrTable kt_dp_branch_dev_info_table[] = {
+    /* Jaguar MCDP50x0 */
+    {KT_CHIP_JAGUAR_5000, KT_FW_STATE_RUN_IROM, {'5', '0', '1', '0', 'I', 'R'}, 6},
+    {KT_CHIP_JAGUAR_5000, KT_FW_STATE_RUN_APP, {'K', 'T', '5', '0', 'X', '0'}, 6},
+    /* Mustang MCDP52x0 */
+    {KT_CHIP_MUSTANG_5200, KT_FW_STATE_RUN_IROM, {'5', '2', '1', '0', 'I', 'R'}, 6},
+    {KT_CHIP_MUSTANG_5200, KT_FW_STATE_RUN_APP, {'K', 'T', '5', '2', 'X', '0'}, 6},
+};
+
+static KtDpDevInfo dp_dev_infos[MAX_DEV_NUM];
+static const gchar *kt_dp_fw_run_state_strs[KT_FW_STATE_NUM] = {"iROM",
+								"App",
+								"Boot-Code",
+								"Unknown"};
+
```

Comment:
can this be moved into `fu_kinetic_dp_aux_isp_get_fw_run_state_str` instead with a switch/case or a bunch of `if`'s?